kaitai/kaitaistream.h

```diff
@@ -281,6 +282,13 @@ inline auto to_signed(From from) {
     throw std::runtime_error("toSigned: Invalid casting");
 }
 
+template<class Output, class Read>
+inline void type_pun(const Output& output, const Read& read)
+{
+    static_assert(sizeof(Read) == sizeof(Output), "Type sizes don't match");
```

Comment:
be careful, I think Read and Output are both pointers.. so the sizes will match always!
If they were not pointer the 'const' in output doesn't make sense.
consider do it as follow:

You can do either fix the static_assert or use parameters as reference instead of pointer, for instance:

```
inline void type_pun(Output& output, const Read& read)
{
static_assert(sizeof(read) == sizeof(output)>, ....);
std::memcpy(&output, &read, sizeof(output))
}
used as follow:

type_pun(out, t)
```